src/components/periodic-table-page/PeriodicTablePage.test.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from "react";
+import {
+  render,
+  screen,
+  waitForElementToBeRemoved,
+} from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import PeriodicTablePage from "./PeriodicTablePage";
+import { Router } from "react-router-dom";
+import { createMemoryHistory } from "history";
+import "hammerjs";
+
+test("should render periodic table and able to see details of element", async () => {
```

Comment:
Should this be split into two tests to keep the tests atomic?

Reply:
How you want these tests to be atomic? Suggestions?

Reply:
Something like:
```
describe("should show periodic table and element info", () => {
  test("should render the periodic table", () => {
    // test
  })
  test("should open element-info modal", () => {
    // test
  })
})
```

How does this seem?

Reply:
Indeed seems good and I thought of writing it but turns out it will duplicate the steps from above test to open modal and another test will only contain single line `expect(screen.queryByTestId("overlay")).toBeInTheDocument();`

That's why I thought of combining test as consider if somehow the modal doesn't opens due to any reason, then above test will automatically fail and the existence of second test won't make any difference.

So IMHO just to avoid duplicate tests I thought of creating a single test to verify flow :)